shared/src/schema/user.schema.ts

```diff
@@ -0,0 +1,29 @@
+import { string } from 'zod'
+
+export const VALID_ALIAS_REGEX = /^(?=.{3,20}$)[a-zA-Z0-9]+(?:[_-][a-zA-Z0-9]+?)*$/
+
+const RESERVED_ALIAS = [
+  'admin',
+  'email',
+  'gast',
+  'gdd',
+  'gradido',
+  'guest',
+  'home',
+  'root',
+  'support',
+  'temp',
+  'tmp',
+  'tmp',
+  'user',
+  'usr',
+  'var',
+]
+
+export const aliasSchema = string()
+  .min(3, 'Given alias is too short')
+  .max(20, 'Given alias is too long')
```

Comment:
is "max(20..." equals >20 or >=20?
i simply ask to ensure the correct edges with the correct message.